File: src/math/divisors.c
/* This is a 79-character-wide ASCII C source-code text file.                */
/*=======|=========|=========|=========|=========|=========|=========|=======*/

/****************************************************************************\
 * divisors.c                                                               *
 * This is a 78-character-wide ASCII C source-code text file.               *
 * Prints all of the positive-integer divisors of a given positive integer. *
 *                                                                          *
 * Author: Robbie Hatley                                                    *
 * Date Written: Wed Feb 17, 2016.                                          *
\****************************************************************************/

#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>

int main(int Fred, char* Ethel[])
{
   unsigned long  i       = 0;
   unsigned long  Number  = 0;
   
   if (Fred != 2) {return 666;}

   Number = strtoull(Ethel[1], NULL, 10);

   for ( i = 1 ; i <= Number ; ++i )
   {
      if (0 == Number % i)
      {
         printf(" %lu", i);
      }
   }
   return 0;
}

File: scripts/math/divisors.pl
#!/usr/bin/perl

# This is a 120-character-wide Unicode UTF-8 Perl-source-code text file with hard Unix line breaks ("\x0A").
# ¡Hablo Español! Говорю Русский. Björt skjöldur. ॐ नमो भगवते वासुदेवाय.    看的星星，知道你是爱。 麦藁雪、富士川町、山梨県。
# =======|=========|=========|=========|=========|=========|=========|=========|=========|=========|=========|=========|

##################################################################
# "divisors.pl"
# Prints all of the divisors of the given positive integer.
# Author: Robbie Hatley.
# Edit history:
#    Mon Apr 27, 2015 - Wrote it.
##################################################################

use v5.32;
use strict;
use warnings;
use utf8;
use warnings FATAL => "utf8";

# Main:
{
   my  $i       = 0;
   my  $Number  = 0;
   my  @divs    = ();
   my  $divs    = 0;

   if (1 != scalar @ARGV) {return 666;}

   $Number = 0 + $ARGV[0];

   for ( $i = 1 ; $i <= $Number ; ++$i )
   {
      if (0 == $Number % $i)
      {
         push @divs, $i;
         ++$divs;
      }
   }
   say "$Number has $divs divisors:";
   $, = ' ';
   say @divs;
   exit 0;
}

